mimic/test/test_core.py

```diff
@@ -141,6 +142,102 @@ class brokenApiMock(object):
         with self.assertRaises(TypeError):
             MimicCore(Clock(), [brokenApiMock()])
 
+    def test_load_duplicate_api_uuid(self):
+        eeapi = make_example_external_api(
+            self,
+            name=self.eeapi_name,
+        )
+        eeapi2 = make_example_external_api(
+            self,
+            name="alternate " + self.eeapi_name
+        )
+        eeapi2.uuid_key = eeapi.uuid_key
+
+        with self.assertRaises(ValueError):
+            MimicCore(Clock(), [eeapi, eeapi2])
+
+    def test_load_duplicate_api_name(self):
+        eeapi = make_example_external_api(
+            self,
+            name=self.eeapi_name,
+        )
+        eeapi2 = make_example_external_api(
+            self,
+            name=self.eeapi_name
+        )
+        # Note: make_example_external_api makes the UUID
+        # to be uuid-<name> so we need to change it for eeapi2
+        eeapi2.uuid_key = str(uuid.uuid4())
+
+        with self.assertRaises(ValueError):
+            MimicCore(Clock(), [eeapi, eeapi2])
+
+    def test_remove_api(self):
+        """
+        Validate that an API can be removed
+        """
+        eeapi = make_example_external_api(
+            self,
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+        self.assertIsNotNone(eeapi)
+
+        template_ids = [ept.id_key for ept in eeapi.list_templates()]
+        for template_id in template_ids:
+            eeapi.remove_template(template_id)
+
+        core = MimicCore(Clock(), [eeapi])
+        core.remove_external_api(
+            eeapi.uuid_key
+        )
+
+    def test_remove_api_invalid(self):
+        """
+        Validate API removal fails when it does not find the
```

Comment:
again apropos of https://jml.io/pages/test-docstrings.html you can just delete the word "Validate" here :)